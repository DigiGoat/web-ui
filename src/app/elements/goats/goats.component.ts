import type { HttpErrorResponse } from '@angular/common/http';
import { Goat } from 'src/app/services/goat/goat.service';

import { Component, Input, OnInit } from '@angular/core';


import { ActivatedRoute } from '@angular/router';
import type { Observable } from 'rxjs';
import { PlatformService } from '../../services/platform/platform.service';


@Component({
  selector: 'app-goats',
  templateUrl: './goats.component.html',
  styleUrls: ['./goats.component.scss']
})
export class GoatsComponent implements OnInit {
  public err?: HttpErrorResponse;
  public noGoats = false;
  public activeGoatIndex = -1;
  public prerender = false;
  public bot = false;
  public searchParam?: string;
  constructor(public route: ActivatedRoute, private platformService: PlatformService) {
  }
  public goats?: Goat[];
  @Input({ required: true, alias: 'goats' }) getter!: Observable<Goat[]>;
  @Input({ required: true }) name!: string;
  ngOnInit() {
    this.prerender = this.platformService.isServer;
    this.bot = this.platformService.isBot;
    this.searchParam = this.route.snapshot.params['goat'];
    this.getter.subscribe({
      next: goats => {
        if (!goats.length) {
          this.noGoats = true;
        }
        this.goats = goats;
        if (this.searchParam) {
          this.activeGoatIndex = this.goats?.findIndex(goat => [goat.nickname, goat.name, goat.normalizeId].includes(this.searchParam));
        }
      },
      error: err => this.err = err
    });
  }
}
