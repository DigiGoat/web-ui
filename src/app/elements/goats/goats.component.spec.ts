import { of } from 'rxjs/internal/observable/of';
import { throwError } from 'rxjs/internal/observable/throwError';

import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ActivatedRoute } from '@angular/router';
import type { PlatformService } from '../../services/platform/platform.service';
import { GoatsComponent } from './goats.component';
jest.mock('@angular/router');
const goats = [{
  nickname: 'TEST_NICKNAME_1',
  name: 'TEST_NAME_1',
  description: 'TEST_DESCRIPTION_1',
  normalizeId: 'TEST_ID_1',
  dateOfBirth: 'TEST_BIRTH_1'
}, {
  nickname: 'TEST_NICKNAME_2',
  name: 'TEST_NAME_2',
  description: 'TEST_DESCRIPTION_2',
  normalizeId: 'TEST_ID_2',
  dateOfBirth: 'TEST_BIRTH_2'
}];
describe('GoatsComponent', () => {
  let component: GoatsComponent;
  let fixture: ComponentFixture<GoatsComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [GoatsComponent],
      providers: [ActivatedRoute],
    })
      .compileComponents();
    fixture = TestBed.createComponent(GoatsComponent);
    component = fixture.componentInstance;
    component.name = 'TEST GOATS';
  });
  describe('Without a searchParam', () => {
    beforeEach(() => {
      component['route'] = {
        snapshot: {
          params: {
          }
        }
      } as unknown as ActivatedRoute;
    });
    describe('Without Goat\'s', () => {
      let html: HTMLElement;
      beforeEach(() => {
        component.getter = of([]);
        fixture.detectChanges();
        html = fixture.nativeElement;
      });
      it('should create', () => {
        expect(component).toBeTruthy();
      });
      it('should set noGoats to true', () => {
        expect(component.noGoats).toBe(true);
      });
      it('should not display goats', () => {
        const element = html.querySelector('[test-id=goats]');
        expect(element).toBeFalsy();
      });
      it('should not display an error', () => {
        const element = html.querySelector('[test-id=error]');
        expect(element).toBeFalsy();
      });
      it('should display no-goats', () => {
        const element = html.querySelector('[test-id=no-goats]');
        expect(element).toBeTruthy();
      });
      it('should display no goats title', () => {
        const element = html.querySelector('[test-id=no-goats-title]');
        expect(element?.innerHTML).toBe('No TEST GOATS Found');
      });
      it('should display no goats prompt', () => {
        const element = html.querySelector('[test-id=no-goats-prompt]');
        expect(element?.innerHTML).toBe('Please Try Again Later.');
      });
      it('should match the snapshot', () => {
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
      });
    });
    describe('With a Error', () => {
      let html: HTMLElement;
      beforeEach(() => {
        component.getter = throwError(() => { return { status: -1, statusText: 'TEST_ERROR' }; });
        fixture.detectChanges();
        html = fixture.nativeElement;
      });
      it('should create', () => {
        expect(component).toBeTruthy();
      });
      it('should set err', () => {
        expect(component.err).toEqual({ status: -1, statusText: 'TEST_ERROR' });
      });
      it('should not display goats', () => {
        const element = html.querySelector('[test-id=goats]');
        expect(element).toBeFalsy();
      });
      it('should not display no-goats', () => {
        const element = html.querySelector('[test-id=no-goats]');
        expect(element).toBeFalsy();
      });
      it('should display error', () => {
        const element = html.querySelector('[test-id=error]');
        expect(element).toBeTruthy();
      });
      it('should display an error title', () => {
        const element = html.querySelector('[test-id=error-title]');
        expect(element?.innerHTML).toBe('Failed To Load TEST GOATS With Error: -1 - TEST_ERROR');
      });
      it('should display an error prompt', () => {
        const element = html.querySelector('[test-id=error-prompt]');
        expect(element?.innerHTML).toBe('Please Check Your Connection And Try Again Later.');
      });
      it('should match the snapshot', () => {
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
      });
    });
    describe('With Goat\'s', () => {
      let html: HTMLElement;
      beforeEach(() => {
        component.getter = of(goats);
        fixture.detectChanges();
        html = fixture.nativeElement;
      });
      it('should create', () => {
        expect(component).toBeTruthy();
      });
      it('should not display no-goats', () => {
        const element = html.querySelector('[test-id=no-goats]');
        expect(element).toBeFalsy();
      });
      it('should not display an error', () => {
        const element = html.querySelector('[test-id=error]');
        expect(element).toBeFalsy();
      });
      it('should display goats', () => {
        const element = html.querySelector('[test-id=goats]');
        expect(element).toBeTruthy();
      });
      it('should display a title', () => {
        const element = html.querySelector('[test-id=name]');
        expect(element?.innerHTML).toBe('TEST GOATS');
      });
      it('should display a description', () => {
        const element = html.querySelector('[test-id=prompt]');
        expect(element?.innerHTML).toBe('Click on a TEST GOAT Below For More Info');
      });
      it('should display the goats', () => {
        const elements = html.querySelectorAll('[test-id=goat-cards]');
        expect(elements).toBeTruthy();
        expect(elements).toHaveLength(goats.length);
      });
      it('should not display placeholders', () => {
        const elements = html.querySelector('[test-id=goat-placeholders]');
        expect(elements).toBeFalsy();
      });
      it('should match the snapshot', () => {
        expect(fixture).toMatchSnapshot();
      });
      describe('during loading', () => {
        beforeEach(() => {
          component['goats'] = [];
          fixture.detectChanges();
          html = fixture.nativeElement;
        });
        it('should create', () => {
          expect(component).toBeTruthy();
        });
        it('should not display no-goats', () => {
          const element = html.querySelector('[test-id=no-goats]');
          expect(element).toBeFalsy();
        });
        it('should not display an error', () => {
          const element = html.querySelector('[test-id=error]');
          expect(element).toBeFalsy();
        });
        it('should display goats', () => {
          const element = html.querySelector('[test-id=goats]');
          expect(element).toBeTruthy();
        });
        it('should display a title', () => {
          const element = html.querySelector('[test-id=name]');
          expect(element?.innerHTML).toBe('TEST GOATS');
        });
        it('should display a description', () => {
          const element = html.querySelector('[test-id=prompt]');
          expect(element?.innerHTML).toBe('Click on a TEST GOAT Below For More Info');
        });
        it('should not display the goats', () => {
          const elements = html.querySelector('[test-id=goat-cards]');
          expect(elements).toBeFalsy();
        });
        it('should display the placeholders', () => {
          const elements = html.querySelectorAll('[test-id=placeholder-cards]');
          expect(elements).toBeTruthy();
          expect(elements).toHaveLength(8);
        });
        it('should display the placeholder cards', () => {
          const elements = html.querySelectorAll('[test-id=placeholder-card]');
          expect(elements).toBeTruthy();
          expect(elements).toHaveLength(8);
        });
        it('should match the snapshot', () => {
          expect(fixture).toMatchSnapshot();
        });
      });
    });
  });
  describe('With a valid searchParam', () => {
    beforeEach(() => {
      component['route'] = {
        snapshot: {
          params: {
            goat: 'TEST_ID_2'
          }
        }
      } as unknown as ActivatedRoute;
    });
    describe('With Goat\'s', () => {
      beforeEach(() => {
        component.getter = of(goats);
      });
      describe('should render', () => {
        let html: HTMLElement;
        beforeEach(() => {
          fixture.detectChanges();
          html = fixture.nativeElement;
        });
        it('should not display no-goats', () => {
          const element = html.querySelector('[test-id=no-goats]');
          expect(element).toBeFalsy();
        });
        it('should not display an error', () => {
          const element = html.querySelector('[test-id=error]');
          expect(element).toBeFalsy();
        });
        it('should display goats container', () => {
          const element = html.querySelector('[test-id=goats]');
          expect(element).toBeTruthy();
        });
        it('should create', () => {
          expect(component).toBeTruthy();
        });
        describe('during loading', () => {
          beforeEach(() => {
            component['goats'] = [];
            fixture.detectChanges();
            html = fixture.nativeElement;
          });
          it('should create', () => {
            expect(component).toBeTruthy();
          });
          it('should not display no-goats', () => {
            const element = html.querySelector('[test-id=no-goats]');
            expect(element).toBeFalsy();
          });
          it('should not display an error', () => {
            const element = html.querySelector('[test-id=error]');
            expect(element).toBeFalsy();
          });
          it('should display goats', () => {
            const element = html.querySelector('[test-id=goats]');
            expect(element).toBeTruthy();
          });
          it('should display a title', () => {
            const element = html.querySelector('[test-id=name]');
            expect(element?.innerHTML).toBe('TEST GOATS');
          });
          it('should display a description', () => {
            const element = html.querySelector('[test-id=prompt]');
            expect(element?.innerHTML).toBe('Click on a TEST GOAT Below For More Info');
          });
          it('should not display the goats', () => {
            const elements = html.querySelector('[test-id=goat-cards]');
            expect(elements).toBeFalsy();
          });
          it('should display the placeholders', () => {
            const elements = html.querySelectorAll('[test-id=placeholder-cards]');
            expect(elements).toBeTruthy();
            expect(elements).toHaveLength(8);
          });
          it('should display the placeholder cards', () => {
            const elements = html.querySelectorAll('[test-id=placeholder-card]');
            expect(elements).toBeTruthy();
            expect(elements).toHaveLength(8);
          });
          it('should match the snapshot', () => {
            expect(fixture).toMatchSnapshot();
          });
        });
      });
      describe('On the server', () => {
        let html: HTMLElement;
        beforeEach(() => {
          component['platformService'] = {
            isServer: true,
            isBrowser: false,
            isBot: false
          } as PlatformService;
          fixture.detectChanges();
          html = fixture.nativeElement;
        });
        it('should not display the browser <div>', () => {
          const element = html.querySelector('[test-id=browser]');
          expect(element).toBeFalsy();
        });
        it('should display the server <div>', () => {
          const element = html.querySelector('[test-id=server]');
          expect(element).toBeTruthy();
        });
        it('should display a goat card', () => {
          const element = html.querySelector('[test-id=server-card]');
          expect(element).toBeTruthy();
        });
        it('should match the snapshot', () => {
          expect(fixture).toMatchSnapshot();
        });
      });
      describe('With a bot', () => {
        let html: HTMLElement;
        beforeEach(() => {
          component['platformService'] = {
            isServer: false,
            isBrowser: true,
            isBot: true
          } as PlatformService;
          fixture.detectChanges();
          html = fixture.nativeElement;
        });
        it('should not display the browser <div>', () => {
          const element = html.querySelector('[test-id=browser]');
          expect(element).toBeFalsy();
        });
        it('should display the server <div>', () => {
          const element = html.querySelector('[test-id=server]');
          expect(element).toBeTruthy();
        });
        it('should display a goat card', () => {
          const element = html.querySelector('[test-id=server-card]');
          expect(element).toBeTruthy();
        });
        it('should match the snapshot', () => {
          expect(fixture).toMatchSnapshot();
        });
      });
      describe('In a browser', () => {
        let html: HTMLElement;
        beforeEach(() => {
          component['platformService'] = {
            isServer: false,
            isBrowser: true,
            isBot: false
          } as PlatformService;
          fixture.detectChanges();
          html = fixture.nativeElement;
        });
        it('should not display the server <div>', () => {
          const element = html.querySelector('[test-id=server]');
          expect(element).toBeFalsy();
        });
        it('should display the browser <div>', () => {
          const element = html.querySelector('[test-id=browser]');
          expect(element).toBeTruthy();
        });
        it('should display a page name', () => {
          const element = html.querySelector('[test-id=name]');
          expect(element?.innerHTML).toBe('TEST GOATS');
        });
        it('should display a page prompt', () => {
          const element = html.querySelector('[test-id=prompt]');
          expect(element?.innerHTML).toBe('Click on a TEST GOAT Below For More Info');
        });
        it('should not display the not found modal', () => {
          const element = html.querySelector('[test-id=not-found-modal]');
          expect(element).toBeFalsy();
        });
        it('should display goat cards', () => {
          const elements = html.querySelectorAll('[test-id=goat-cards]');
          expect(elements).toBeTruthy();
          expect(elements).toHaveLength(goats.length);
        });
        it('should display a goat card for each goat', () => {
          const elements = html.querySelectorAll('[test-id=goat-card]');
          expect(elements).toBeTruthy();
          expect(elements).toHaveLength(goats.length);
        });
        it('should display one goat modal', () => {
          const elements = html.querySelectorAll('[test-id=goat-modal]');
          expect(elements).toBeTruthy();
          expect(elements).toHaveLength(1);
        });
        it('should match the snapshot', () => {
          expect(fixture).toMatchSnapshot();
        });
      });
    });
  });
  describe('With an invalid searchParam', () => {
    beforeEach(() => {
      component['route'] = {
        snapshot: {
          params: {
            goat: 'INVALID_ID'
          }
        }
      } as unknown as ActivatedRoute;
    });
    describe('Without Goat\'s', () => {
      let html: HTMLElement;
      beforeEach(() => {
        component.getter = of([]);
        fixture.detectChanges();
        html = fixture.nativeElement;
      });
      it('should create', () => {
        expect(component).toBeTruthy();
      });
      it('should set noGoats to true', () => {
        expect(component.noGoats).toBe(true);
      });
      it('should not display goats', () => {
        const element = html.querySelector('[test-id=goats]');
        expect(element).toBeFalsy();
      });
      it('should not display an error', () => {
        const element = html.querySelector('[test-id=error]');
        expect(element).toBeFalsy();
      });
      it('should display no-goats', () => {
        const element = html.querySelector('[test-id=no-goats]');
        expect(element).toBeTruthy();
      });
      it('should display no goats title', () => {
        const element = html.querySelector('[test-id=no-goats-title]');
        expect(element?.innerHTML).toBe('No TEST GOATS Found');
      });
      it('should display no goats prompt', () => {
        const element = html.querySelector('[test-id=no-goats-prompt]');
        expect(element?.innerHTML).toBe('Please Try Again Later.');
      });
      it('should match the snapshot', () => {
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
      });
    });
    describe('With a Error', () => {
      let html: HTMLElement;
      beforeEach(() => {
        component.getter = throwError(() => { return { status: -1, statusText: 'TEST_ERROR' }; });
        fixture.detectChanges();
        html = fixture.nativeElement;
      });
      it('should create', () => {
        expect(component).toBeTruthy();
      });
      it('should set err', () => {
        expect(component.err).toEqual({ status: -1, statusText: 'TEST_ERROR' });
      });
      it('should not display goats', () => {
        const element = html.querySelector('[test-id=goats]');
        expect(element).toBeFalsy();
      });
      it('should not display no-goats', () => {
        const element = html.querySelector('[test-id=no-goats]');
        expect(element).toBeFalsy();
      });
      it('should display error', () => {
        const element = html.querySelector('[test-id=error]');
        expect(element).toBeTruthy();
      });
      it('should display an error title', () => {
        const element = html.querySelector('[test-id=error-title]');
        expect(element?.innerHTML).toBe('Failed To Load TEST GOATS With Error: -1 - TEST_ERROR');
      });
      it('should display an error prompt', () => {
        const element = html.querySelector('[test-id=error-prompt]');
        expect(element?.innerHTML).toBe('Please Check Your Connection And Try Again Later.');
      });
      it('should match the snapshot', () => {
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
      });
    });
    describe('With Goat\'s', () => {
      let html: HTMLElement;
      beforeEach(() => {
        component.getter = of(goats);
        fixture.detectChanges();
        html = fixture.nativeElement;
      });
      it('should create', () => {
        expect(component).toBeTruthy();
      });
      it('should not display no-goats', () => {
        const element = html.querySelector('[test-id=no-goats]');
        expect(element).toBeFalsy();
      });
      it('should not display an error', () => {
        const element = html.querySelector('[test-id=error]');
        expect(element).toBeFalsy();
      });
      it('should display goats', () => {
        const element = html.querySelector('[test-id=goats]');
        expect(element).toBeTruthy();
      });
      it('should display a title', () => {
        const element = html.querySelector('[test-id=name]');
        expect(element?.innerHTML).toBe('TEST GOATS');
      });
      it('should display a description', () => {
        const element = html.querySelector('[test-id=prompt]');
        expect(element?.innerHTML).toBe('Click on a TEST GOAT Below For More Info');
      });
      it('should display the goats', () => {
        const elements = html.querySelectorAll('[test-id=goat-cards]');
        expect(elements).toBeTruthy();
        expect(elements).toHaveLength(goats.length);
      });
      it('should not display placeholders', () => {
        const elements = html.querySelector('[test-id=goat-placeholders]');
        expect(elements).toBeFalsy();
      });
      it('should display the goat not found modal', () => {
        const elements = html.querySelector('[test-id=not-found-modal]');
        expect(elements).toBeTruthy();

      });
      it('should match the snapshot', () => {
        expect(fixture).toMatchSnapshot();
      });
      describe('during loading', () => {
        beforeEach(() => {
          component['goats'] = [];
          fixture.detectChanges();
          html = fixture.nativeElement;
        });
        it('should create', () => {
          expect(component).toBeTruthy();
        });
        it('should not display no-goats', () => {
          const element = html.querySelector('[test-id=no-goats]');
          expect(element).toBeFalsy();
        });
        it('should not display an error', () => {
          const element = html.querySelector('[test-id=error]');
          expect(element).toBeFalsy();
        });
        it('should display goats', () => {
          const element = html.querySelector('[test-id=goats]');
          expect(element).toBeTruthy();
        });
        it('should display a title', () => {
          const element = html.querySelector('[test-id=name]');
          expect(element?.innerHTML).toBe('TEST GOATS');
        });
        it('should display a description', () => {
          const element = html.querySelector('[test-id=prompt]');
          expect(element?.innerHTML).toBe('Click on a TEST GOAT Below For More Info');
        });
        it('should not display the goats', () => {
          const elements = html.querySelector('[test-id=goat-cards]');
          expect(elements).toBeFalsy();
        });
        it('should display the placeholders', () => {
          const elements = html.querySelectorAll('[test-id=placeholder-cards]');
          expect(elements).toBeTruthy();
          expect(elements).toHaveLength(8);
        });
        it('should display the placeholder cards', () => {
          const elements = html.querySelectorAll('[test-id=placeholder-card]');
          expect(elements).toBeTruthy();
          expect(elements).toHaveLength(8);
        });
        it('should match the snapshot', () => {
          expect(fixture).toMatchSnapshot();
        });
      });
    });
  });
});
