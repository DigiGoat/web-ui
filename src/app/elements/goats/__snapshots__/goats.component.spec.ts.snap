// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoatsComponent With a valid searchParam With Goat's In a browser should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  getter={[Function Observable]}
  goats={[Function Array]}
  name={[Function String]}
  noGoats="false"
  platformService={[Function Object]}
  prerender="false"
  route={[Function Object]}
  searchParam={[Function String]}
>
  <div
    class="container"
    test-id="goats"
  >
    <div
      test-id="browser"
    >
      <h1
        class="text-center"
        test-id="name"
      >
        TEST GOATS
      </h1>
      <h5
        class="text-center"
        test-id="prompt"
      >
        Click on a TEST GOAT Below For More Info
      </h5>
      <div
        class="row row-cols-1 row-cols-md-2 row-cols-xl-3 row-cols-xxl-4 align-items-center justify-content-center"
      >
        <div
          class="col py-3"
          test-id="goat-cards"
        >
          <app-goat-card
            test-id="goat-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="goat-cards"
        >
          <app-goat-card
            test-id="goat-card"
          />
          <app-modal-goat
            test-id="goat-modal"
          />
        </div>
      </div>
    </div>
  </div>
</app-goats>
`;

exports[`GoatsComponent With a valid searchParam With Goat's On the server should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  getter={[Function Observable]}
  goats={[Function Array]}
  name={[Function String]}
  noGoats="false"
  platformService={[Function Object]}
  prerender={[Function Boolean]}
  route={[Function Object]}
  searchParam={[Function String]}
>
  <div
    class="container"
    test-id="goats"
  >
    <div
      test-id="server"
    >
      <app-modal-goat-card
        test-id="server-card"
      />
    </div>
  </div>
</app-goats>
`;

exports[`GoatsComponent With a valid searchParam With Goat's With a bot should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot={[Function Boolean]}
  getter={[Function Observable]}
  goats={[Function Array]}
  name={[Function String]}
  noGoats="false"
  platformService={[Function Object]}
  prerender="false"
  route={[Function Object]}
  searchParam={[Function String]}
>
  <div
    class="container"
    test-id="goats"
  >
    <div
      test-id="server"
    >
      <app-modal-goat-card
        test-id="server-card"
      />
    </div>
  </div>
</app-goats>
`;

exports[`GoatsComponent With a valid searchParam With Goat's should render during loading should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  getter={[Function Observable]}
  goats={[Function Array]}
  name={[Function String]}
  noGoats="false"
  platformService={[Function PlatformService]}
  prerender="false"
  route={[Function Object]}
  searchParam={[Function String]}
>
  <div
    class="container"
    test-id="goats"
  >
    <div
      test-id="browser"
    >
      <h1
        class="text-center"
        test-id="name"
      >
        TEST GOATS
      </h1>
      <h5
        class="text-center"
        test-id="prompt"
      >
        Click on a TEST GOAT Below For More Info
      </h5>
      <div
        class="row row-cols-1 row-cols-md-2 row-cols-xl-3 row-cols-xxl-4 align-items-center justify-content-center"
      >
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
      </div>
    </div>
  </div>
</app-goats>
`;

exports[`GoatsComponent With an invalid searchParam With Goat's during loading should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  getter={[Function Observable]}
  goats={[Function Array]}
  name={[Function String]}
  noGoats="false"
  platformService={[Function PlatformService]}
  prerender="false"
  route={[Function Object]}
  searchParam={[Function String]}
>
  <div
    class="container"
    test-id="goats"
  >
    <div
      test-id="browser"
    >
      <h1
        class="text-center"
        test-id="name"
      >
        TEST GOATS
      </h1>
      <h5
        class="text-center"
        test-id="prompt"
      >
        Click on a TEST GOAT Below For More Info
      </h5>
      <app-modal-goat
        test-id="not-found-modal"
      />
      <div
        class="row row-cols-1 row-cols-md-2 row-cols-xl-3 row-cols-xxl-4 align-items-center justify-content-center"
      >
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
      </div>
    </div>
  </div>
</app-goats>
`;

exports[`GoatsComponent With an invalid searchParam With Goat's should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  getter={[Function Observable]}
  goats={[Function Array]}
  name={[Function String]}
  noGoats="false"
  platformService={[Function PlatformService]}
  prerender="false"
  route={[Function Object]}
  searchParam={[Function String]}
>
  <div
    class="container"
    test-id="goats"
  >
    <div
      test-id="browser"
    >
      <h1
        class="text-center"
        test-id="name"
      >
        TEST GOATS
      </h1>
      <h5
        class="text-center"
        test-id="prompt"
      >
        Click on a TEST GOAT Below For More Info
      </h5>
      <app-modal-goat
        test-id="not-found-modal"
      />
      <div
        class="row row-cols-1 row-cols-md-2 row-cols-xl-3 row-cols-xxl-4 align-items-center justify-content-center"
      >
        <div
          class="col py-3"
          test-id="goat-cards"
        >
          <app-goat-card
            test-id="goat-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="goat-cards"
        >
          <app-goat-card
            test-id="goat-card"
          />
        </div>
      </div>
    </div>
  </div>
</app-goats>
`;

exports[`GoatsComponent With an invalid searchParam With a Error should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  err={[Function Object]}
  getter={[Function Observable]}
  name={[Function String]}
  noGoats="false"
  platformService={[Function PlatformService]}
  prerender="false"
  route={[Function Object]}
  searchParam={[Function String]}
>
  <div
    test-id="error"
  >
    <h3
      class="text-center"
      test-id="error-title"
    >
      Failed To Load TEST GOATS With Error: -1 - TEST_ERROR
    </h3>
    <h6
      class="text-center"
      test-id="error-prompt"
    >
      Please Check Your Connection And Try Again Later.
    </h6>
  </div>
</app-goats>
`;

exports[`GoatsComponent With an invalid searchParam Without Goat's should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  getter={[Function Observable]}
  goats={[Function Array]}
  name={[Function String]}
  noGoats={[Function Boolean]}
  platformService={[Function PlatformService]}
  prerender="false"
  route={[Function Object]}
  searchParam={[Function String]}
>
  <div
    test-id="no-goats"
  >
    <h3
      class="text-center"
      test-id="no-goats-title"
    >
      No TEST GOATS Found
    </h3>
    <h6
      class="text-center"
      test-id="no-goats-prompt"
    >
      Please Try Again Later.
    </h6>
  </div>
</app-goats>
`;

exports[`GoatsComponent Without a searchParam With Goat's during loading should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  getter={[Function Observable]}
  goats={[Function Array]}
  name={[Function String]}
  noGoats="false"
  platformService={[Function PlatformService]}
  prerender="false"
  route={[Function Object]}
  searchParam="undefined"
>
  <div
    class="container"
    test-id="goats"
  >
    <div
      test-id="browser"
    >
      <h1
        class="text-center"
        test-id="name"
      >
        TEST GOATS
      </h1>
      <h5
        class="text-center"
        test-id="prompt"
      >
        Click on a TEST GOAT Below For More Info
      </h5>
      <div
        class="row row-cols-1 row-cols-md-2 row-cols-xl-3 row-cols-xxl-4 align-items-center justify-content-center"
      >
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="placeholder-cards"
        >
          <app-goat-card
            test-id="placeholder-card"
          />
        </div>
      </div>
    </div>
  </div>
</app-goats>
`;

exports[`GoatsComponent Without a searchParam With Goat's should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  getter={[Function Observable]}
  goats={[Function Array]}
  name={[Function String]}
  noGoats="false"
  platformService={[Function PlatformService]}
  prerender="false"
  route={[Function Object]}
  searchParam="undefined"
>
  <div
    class="container"
    test-id="goats"
  >
    <div
      test-id="browser"
    >
      <h1
        class="text-center"
        test-id="name"
      >
        TEST GOATS
      </h1>
      <h5
        class="text-center"
        test-id="prompt"
      >
        Click on a TEST GOAT Below For More Info
      </h5>
      <div
        class="row row-cols-1 row-cols-md-2 row-cols-xl-3 row-cols-xxl-4 align-items-center justify-content-center"
      >
        <div
          class="col py-3"
          test-id="goat-cards"
        >
          <app-goat-card
            test-id="goat-card"
          />
        </div>
        <div
          class="col py-3"
          test-id="goat-cards"
        >
          <app-goat-card
            test-id="goat-card"
          />
        </div>
      </div>
    </div>
  </div>
</app-goats>
`;

exports[`GoatsComponent Without a searchParam With a Error should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  err={[Function Object]}
  getter={[Function Observable]}
  name={[Function String]}
  noGoats="false"
  platformService={[Function PlatformService]}
  prerender="false"
  route={[Function Object]}
  searchParam="undefined"
>
  <div
    test-id="error"
  >
    <h3
      class="text-center"
      test-id="error-title"
    >
      Failed To Load TEST GOATS With Error: -1 - TEST_ERROR
    </h3>
    <h6
      class="text-center"
      test-id="error-prompt"
    >
      Please Check Your Connection And Try Again Later.
    </h6>
  </div>
</app-goats>
`;

exports[`GoatsComponent Without a searchParam Without Goat's should match the snapshot 1`] = `
<app-goats
  activeGoatIndex={[Function Number]}
  bot="false"
  getter={[Function Observable]}
  goats={[Function Array]}
  name={[Function String]}
  noGoats={[Function Boolean]}
  platformService={[Function PlatformService]}
  prerender="false"
  route={[Function Object]}
  searchParam="undefined"
>
  <div
    test-id="no-goats"
  >
    <h3
      class="text-center"
      test-id="no-goats-title"
    >
      No TEST GOATS Found
    </h3>
    <h6
      class="text-center"
      test-id="no-goats-prompt"
    >
      Please Try Again Later.
    </h6>
  </div>
</app-goats>
`;
